types/architect__functions/ws.d.ts

```diff
@@ -0,0 +1,22 @@
+import { HttpResponse } from './http';
+
+// https://arc.codes/docs/en/reference/app.arc/ws (current docs site, lacks detail)
+// https://v6.arc.codes/primitives/ws (previous docs site, more info)
+export interface WebsocketEvent {
+    requestContext: {
+        connectionId: string;
+    };
+    isBase64Encoded: boolean;
+    body: string;
+}
+
+// WebSockets still respond with HTTP
+// (usually just a 200 acknowledging the socket is connected)
+export interface WebSocketHandler {
+    (req: WebsocketEvent): Promise<HttpResponse | undefined>;
+}
+
+export interface ArcWs {
```

Comment:
I suggest adding some "tests" for this in [/types/architect__functions/test](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/architect__functions/test). Note the "tests" don't need to exercise any behavior, just use the types. So they're usually pretty trivial. [This section in readme](https://github.com/DefinitelyTyped/DefinitelyTyped/#my-package-teststs) explains a bit more.

Reply:
Ah that makes sense - I did look at the tests but couldn't see anything actually being tested. I'll finish this off soon.